flo-workflow/src/test/java/com/spotify/flo/InputSyncCompletionTest.java

```diff
@@ -82,12 +84,61 @@ public void shouldNotCompleteTaskBeforeAllInputsAreDone_AsInputListBefore() thro
   @Test
   public void shouldNotCompleteTaskBeforeAllInputsAreDone_AsInputListAfter() throws Exception {
     Task<String> task = Task.named("sync").ofType(String.class)
-        .inputs(() -> Arrays.asList(blocking, failing))
+        .inputs(() -> Arrays.asList(blocking, failing("failed")))
         .process(a -> "should not happen");
 
     awaitBlocked(task);
   }
 
+  @Test
+  public void shouldPropagateExceptionsAsSuppressed() throws Exception {
+    Task<String> task = Task.named("failing").ofType(String.class)
+        .input(() -> failing("failed 1"))
+        .input(() -> failing("failed 2"))
+        .process((a, b) -> "should not happen");
+
+    AwaitValue<Throwable> eval = new AwaitValue<>();
+    context.evaluate(task).onFail(eval);
+
+    Throwable throwable = eval.awaitAndGet();
+
+    assertThat(throwable, instanceOf(RuntimeException.class));
+    assertThat(throwable.getMessage(), is("failed 2"));
+    assertThat(throwable.getSuppressed().length, is(1));
+
+    throwable = throwable.getSuppressed()[0];
+    assertThat(throwable.getMessage(), is("failed 1"));
+  }
+
+  @Test
+  public void shouldPropagateExceptionsAsSuppressedFromMultipleLevels() throws Exception {
+    Task<String> taskLevel2 = Task.named("failing-2").ofType(String.class)
+        .input(() -> failing("failed 1"))
+        .input(() -> failing("failed 2"))
+        .process((a, b) -> "should not happen");
+
+    Task<String> task = Task.named("failing-1").ofType(String.class)
+        .input(() -> taskLevel2)
+        .input(() -> failing("failed 3"))
+        .process((a, b) -> "should not happen");
+
+    AwaitValue<Throwable> eval = new AwaitValue<>();
+    context.evaluate(task).onFail(eval);
+
+    Throwable throwable = eval.awaitAndGet();
+
+    assertThat(throwable, instanceOf(RuntimeException.class));
+    assertThat(throwable.getMessage(), is("failed 3"));
+    assertThat(throwable.getSuppressed().length, is(1));
+
+    throwable = throwable.getSuppressed()[0];
+    assertThat(throwable.getMessage(), is("failed 2"));
+    assertThat(throwable.getSuppressed().length, is(1));
```

Comment:
Same as above.

Reply:
This is already done in the following lines. Sense for the two other comments above.

Reply:
My bad